src/main/java/org/broadinstitute/hellbender/tools/spark/sv/AlignedAssemblyOrExcuse.java

```diff
@@ -324,6 +278,68 @@ private static BwaMemAlignment readAlignment(final Input input) {
                 mateRefId, mateRefStart, templateLen);
     }
 
+    public int getAssemblyId() {
+        return assemblyId;
+    }
+
+    /**
+     * Either this is null, or the assembly and list of alignments is null.
+     */
+    public String getErrorMessage() {
+        return errorMessage;
+    }
+
+    public boolean isNotFailure() {
+        return errorMessage==null;
+    }
+
+    public FermiLiteAssembly getAssembly() {
+        return assembly;
+    }
+
+    /**
+     * List is equal in length to the number of contigs in the assembly.
+     */
+    public List<List<BwaMemAlignment>> getContigAlignments() {
+        return contigAlignments;
+    }
+
+    private Stream<SAMRecord> toSAMStreamForAlignmentsOfThisAssembly(final SAMFileHeader header, final List<String> refNames) {
+        Utils.validate(isNotFailure(), "Can't stream SAM records from a failed assembly.");
+        return IntStream.range(0, contigAlignments.size()).boxed()
+                .flatMap(contigIdx ->
+                        toSAMStreamForOneContig(header,refNames, assemblyId,contigIdx,
```

Comment:
just to annoy you: please remove the space between `refNames, assemblyId` :smirk: